src/util/memory.rs

```diff
@@ -147,6 +147,8 @@ pub fn handle_mmap_error<VM: VMBinding>(error: Error, tls: VMThread) -> ! {
     match error.kind() {
         // From Rust nightly 2021-05-12, we started to see Rust added this ErrorKind.
         ErrorKind::OutOfMemory => {
+            #[cfg(target_os = "linux")]
+            info!("{}", get_process_memory_maps());
```

Comment:
1.  It is better if we make the call regardless of the current OS, but we may implement the function platform-specific.  At this moment, we can make the function do nothing (or return empty string or "(unavailable)") for systems other than Linux.
2.  We should just print using `eprint!` or `eprintln!` (or do the printing inside `get_process_memory_maps`, or introduce a `print_process_memory_maps`).  It will be very helpful to have the information when the release build fails.  And if it fails, it doesn't matter whether we print more things to the console or less.

Reply:
Sure

Reply:
Ideally we want more information (what address was MMTk trying to `mmap`), but our current internal API for all this is quite wonky. And it seems like `handle_mmap_failure` is only used for metadata when `mmap` can fail in normal space allocation as well